contracts/governance/DESIGN.md

```diff
@@ -0,0 +1,59 @@
+# Axelar Governance Protocol
+
+## Multisig
+Central to the Axelar Governance Protocol is a Multisig contract, incorporating MultisigBase and Multisig components. This contract maintains a list of signer accounts and enforces a threshold or minimum number of signers required to execute certain functions. It plays a pivotal role in defining token mint limits at the gateway.
+
+### Voting Procedure
+For each transaction, a voting process is initiated among the signers. Each time a signer casts a vote, the count increments and the act is documented against the signer's address. Upon reaching the threshold, the transaction executes and the voting counter resets.
+
+### Rotation of Signers
+The Multisig contract facilitates a rotation mechanism for signers. Existing signers can refresh the list of signatory addresses and the threshold value using the `rotateSigners` function. Each rotation is tracked by incrementing an epoch.
+
+### Execution of External Contracts
+The Multisig contract extends its purview to external contracts. The `execute` function allows signers to call any external contract. The data for this call encodes the target contract function and arguments, and may include native tokens to accompany the call.
```

Comment:
```suggestion
The Multisig contract extends its purview functionality to include external contract calls. The `execute` function allows signers to call any function on any external contract. The data accompanying this call contains the target contract's function and its associated arguments. It may also include native tokens as part of the transaction.
```